include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -698,45 +719,112 @@ class graph_oriented_layout_design_impl
-                        restore_names(ssg.network, best_lyt);
+                    futures.emplace_back(std::async(
+                        std::launch::async,
+                        [&]() -> std::optional<Lyt>  // return std::optional<layout>
+                        {
+                            if (ssg.frontier_flag)
+                            {
+                                const auto expansion = expand(ssg);
+                                if (expansion.second)
+                                {
+                                    const std::lock_guard<std::mutex> lock(mtx);  // protect access to shared data
+                                    best_lyt = *expansion.second;
+                                    restore_names(ssg.network, best_lyt);
+
+                                    // statistical information
+                                    pst.x_size        = best_lyt.x() + 1;
+                                    pst.y_size        = best_lyt.y() + 1;
+                                    pst.num_gates     = best_lyt.num_gates();
+                                    pst.num_wires     = best_lyt.num_wires();
+                                    pst.num_crossings = best_lyt.num_crossings();
+
+                                    if (ps.return_first)
+                                    {
+                                        return best_lyt;  // return the layout if ps.return_first is true
+                                    }
+                                }
 
-                        // statistical information
-                        pst.x_size        = best_lyt.x() + 1;
-                        pst.y_size        = best_lyt.y() + 1;
-                        pst.num_gates     = best_lyt.num_gates();
-                        pst.num_wires     = best_lyt.num_wires();
-                        pst.num_crossings = best_lyt.num_crossings();
+                                // update costs and frontier
+                                for (const auto& [next, cost] : expansion.first)
+                                {
+                                    if (ssg.cost_so_far.find(next) == ssg.cost_so_far.cend() ||
+                                        cost < ssg.cost_so_far[next])
+                                    {
+                                        ssg.cost_so_far[next] = cost;
+                                        double priority       = cost;
+                                        ssg.frontier.put(next, priority);
+                                    }
+                                }
+                            }
+                            return std::nullopt;  // return no layout found
+                        }));
+                }
 
-                        if (ps.return_first)
-                        {
-                            return best_lyt;
-                        }
+                // Check the futures for the result
+                for (auto& future : futures)
```

Comment:
```suggestion
                for (const auto& future : futures)
```
?

Reply:
The use of `const auto& future` makes `future` a constant reference to a `std::future`, which means you cannot call `get()` because `get()` is a non-const method.